packages/grid/_modules_/grid/hooks/features/columnReorder/useColumnReorder.tsx

```diff
@@ -97,35 +112,32 @@ export const useColumnReorder = (columnsRef: React.RefObject<HTMLDivElement>, ap
     };
   }, []);
 
-  const handleColumnHeaderDragOver = React.useCallback(
-    (event) => {
+  const onColHeaderDragOver = React.useCallback(
+    (event: Event, ref: React.RefObject<HTMLElement>) => {
       event.preventDefault();
       apiRef.current.publishEvent(COL_REORDER_DRAG_OVER_HEADER);
-
-      columnsRef.current!.classList.add(HEADER_CELL_DROP_ZONE_CSS_CLASS);
+      columnsHeaderRef.current = ref.current;
+      columnsHeaderRef.current!.classList.add(HEADER_CELL_DROP_ZONE_CSS_CLASS);
```

Comment:
Can we refactor that to state props...?

Reply:
From what I understand, the class needs to be applied before the next RAF, so it's not possible. @DanailH thoughts?

Reply:
That's correct, also I don't think it's necessary to preserve this in the state as it is only used inside this hook. It doesn't bring any value outside of the `useColumnReorder`.